multirun/process.go

```diff
@@ -59,7 +59,8 @@ func (p *process) run(ctx context.Context) error {
 		}()
 	}
 	done := make(chan struct{})
-	defer close(done) // unblocks the goroutine below when cmd.Wait() returns
+	var wg2 sync.WaitGroup
+	wg2.Add(1)
```

Comment:
I think you are right, because `select` is not deterministic there is a race. The problem happens if the first case is selected.

Could you change the code a bit like so:

```go
var wg2 sync.WaitGroup
defer wg2.Wait() // waiting for terminating go-routine to finish
done := make(chan struct{})
defer close(done) // unblocks the goroutine below when cmd.Wait() returns
wg2.Add(1)
go func() {
    defer wg2.Done()
    // ...
}()
wg.Wait()
return cmd.Wait()
```


BTW, in addition to this fix, we could check for that error that you are getting - it's possible that the program finishes just before SIGTERM is sent. I wonder if it can be done without resorting to checking the error message.

Reply:
thanks for the review

If i understand what you mean
> in addition to this fix, we could check for that error that you are getting - it's possible that the program finishes just before SIGTERM is sent

I already added this check in a separate PR https://github.com/atlassian/bazel-tools/pull/114

Reply:
I don't understand how that happened, but I didn't see that PR. Thanks, merged!

Reply:
ok, I'm not sure if waiting in the `defer` is a good practice or not 🤷 
but as a maintainer if you prefer this then it is ok ;)

Reply:
I've done this a thousand times and it's fine. Nobody ever said it's not ok :)